src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -827,7 +827,7 @@ void SpatialPooler::updateBookeepingVars_(bool learn) {
 
 
 void SpatialPooler::calculateOverlap_(const SDR &input,
-                                      vector<SynapseIdx> &overlaps) const {
+                                      vector<SynapseIdx> &overlaps) {
```

Comment:
SP is state-less, but since the method it calls in Connections is stateful, it needs to, either: 
- SP param to set Connections(timeseries)
  - and SP::reset() ? 
- SP always calls the state-less method
  - then for consideration to have 2 versions in conn: `computeActivity();` //does timeseries; and `computeActivity() const;`

Reply:
SP::calculateOverlap_ is a private method... Why can't it be trusted to not change things that it shouldn't?  Why does it need to be `const` ?

Reply:
ok, it can remain non-const, I meant it for optimization and readability, but guess it'd overcomplicate stuff. 

> SP param to set Connections(timeseries)
   and SP::reset() ?
 
this Q still remains, can SP make use of this feature (timeseries), or why not?

Reply:
I suppose it could make use of this feature, but I'd rather not change the SP, and instead focus on the column pooler which is intended to obsolete it.